web/src/pill_typeahead.ts

```diff
@@ -36,25 +37,42 @@ export function set_up(
         user_source?: () => User[];
         exclude_bots?: boolean;
         update_func?: () => void;
+        help_on_empty_strings?: boolean;
+        display_pill?: (sub: StreamSubscription) => string;
     },
 ): void {
     if (!opts.user && !opts.user_group && !opts.stream) {
         blueslip.error("Unspecified possible item types");
         return;
     }
-    const include_streams = (query: string): boolean =>
-        opts.stream !== undefined && query.trim().startsWith("#");
+
+    opts.help_on_empty_strings ||= false;
+
     const include_user_groups = opts.user_group;
     const include_users = opts.user;
     const exclude_bots = opts.exclude_bots;
 
+    const stream_query: (query: string) => boolean | undefined = (query: string) =>
+        opts.stream && query.trim().startsWith("#");
+
+    const include_streams = include_users || include_user_groups ? stream_query : () => opts.stream;
+
+    let display_pill: (sub: StreamSubscription) => string;
+
+    if (!opts.display_pill) {
+        display_pill = stream_pill.display_pill;
+    } else {
+        display_pill = opts.display_pill;
+    }
```

Comment:
Done.

Reply:
I'm not entirely sure this version is easier to read; probably the `??` operator might be clearest?

Reply:
So also of note is that this function being patched here is a generic function not limited to just streams, and yet we have a function called `display_pill` that is only relevant for those? That doesn't make sense.

Reply:
Do we need a default value at all?

Reply:
> Do we need a default value at all?

No, we use `display_pill` when we call `stream_pill.append`. We can change the `stream_pill.append` function such that it accepts an optional parameter for getting stream pill label and it would provide the default value.

Reply:
> So also of note is that this function being patched here is a generic function not limited to just streams, and yet we have a function called display_pill that is only relevant for those? That doesn't make sense.

We can rename the function so it indicates its stream specific like `stream_display_pill`?

I think we need this function in `pill_typeahead.ts` because `pill_typeahead` calls `stream_pill.append` and we need to mention the display value of pill there. It wouldn't been have a problem if we displayed the same format value for stream pill everywhere, but for user invite modal we just want stream name as the display value and not extra information about subscriber count.

 Also you suggested to have `stream_match_prefix` as a parameter which is also specific to streams so i think a function specific to streams should also be alright.